### PR TITLE
Armor Pieces and Other

### PR DESCRIPTION
**Pull request by OcelotWalrus**
This PR contains:
  * edited `.editorconfig` to make the indent size of YAML files to 2 instead of 4.
  * made player `max health` to '15' instead of '10' buy his default current health is still '10'.
  * added 'map zones' which are displayed in the Coordinates display.
  * added new items:
    * The Leather Jacket
    * The Leather Boots
    * The Old Pants
    * Stretched Bag
    * The Knight Boots
  * added new type of item: `Armor Piece`.
  * added `armor protection` stat that can be added to items that have the `Armor Piece: <type>` type or they will be ignored. The `armor protection` stat is finally written on the save file. The `armor protection` stat makes you loose less damage when an enemy fights you, here is the formula:
  * added an inventory slots feature
  * added `inventory slots` stat that can be added to items that have the `Bag` type or they will be ignored. It will make the player inventory slots higher.
    * damage = random.randint(0, enemy_max_damage) - defend * ( armor_protection * random.randint(.5, .9) )
    * For example: 9 - 3.2 * ( .96 * .6 ) = 9 - 1.8 = 8.2 = 8
                   9 - 3.2 * ( 1.5 * .6 ) = 9 - 2.88 = 6.1 = 6
    * the game engine will round the damage after it is calculated
  * implemented use item in battle class for consumable type items so it is totally automatic
    * image
   * Quick note, all consumables will need to have a `max bonus` attribute. So make sure to set it to 0 if you don't want it to have a bonus.
  * print all available saves when starting the game
    * image
  *  add the 'P' command that lets you choose which items to equip
    * image
  * add the 'T' command that lets you get rid of an item
    * image
  * make so that items you've already taken from the ground don't appears back
    * image
    * image

### CODE REVIEW

**Comment by OcelotWalrus** [comment]
I also have an other idea about the armor pieces:

Adding `swiftness` attributes to all armor pieces and make so that the the more you have of `swiftness`, the more you have a chance to jam an enemy attack. And maybe also add a random `swiftness` so they have a chance to jam your attacks.

Should I code that?

**Comment by lumbar527** [comment]
> I also have an other idea about the armor pieces:
> 
> Adding `swiftness` attributes to all armor pieces and make so that the the more you have of `swiftness`, the more you have a chance to jam an enemy attack. And maybe also add a random `swiftness` so they have a chance to jam your attacks. So maybe `swiftness` could also be attributed to mounts if there is any?
> 
> Should I code that?

Sure, though I'm not entirely sure by what you mean by this:
> And maybe also add a random `swiftness` so they have a chance to jam your attacks.

Does that mean another random variable like `enemy_max_health` or `enemy_damage` that would be one of the enemies stats?

File: battle.py [hunk 1]
```diff
@@ -74,7 +75,8 @@ def fight(player, item):
         while not turn:
             # if enemy is still alive
             if enemy_health > 0:
-                damage = random.randint(0, enemy_max_damage) - defend
+                damage = random.randint(0, enemy_max_damage) - defend * ( armor_protection * round(random.uniform(0.50, 0.90), 1) )
```
Comment (lumbar527) [inline battle.py:78]: What does `random.uniform` do?

File: items.yaml [hunk 2]
```diff
@@ -15,4 +15,16 @@ Axe:
   type: "Weapon"
   damage: 7
   defend: 5
-  description: "An ancient axe found in the dungeons. You wonder what happened to it's previous owner..."
\ No newline at end of file
+  description: "An ancient axe found in the dungeons. You wonder what happened to it's previous owner..."
+Leather Jacket:
+  type: "Armor Piece"
+  armor protection: .6
+  description: "A small jacket that may not protect much but keeps you warm when it's cold."
```
Comment (lumbar527) [inline items.yaml:22]:
```suggestion
  description: "A small jacket that may not protect you much but keeps you warm when it's cold."
```

File: items.yaml [hunk 3]
```diff
@@ -15,4 +15,16 @@ Axe:
...
+Leather Boots:
+  type: "Armor Piece"
+  armor protection: .1
+  description: "These boots may be dirt or wet but they let you have great mobility."
```
Comment (lumbar527) [inline items.yaml:26]:
```suggestion
  description: "While these boots are dirty and wet, they grant you great mobility."
```

I presume they also get a great `swiftness` value.

File: items.yaml [hunk 4]
```diff
@@ -15,4 +15,16 @@ Axe:
...
+Old Pants:
+  type: "Armor Piece"
+  armor protection: .25
+  description: "An old pants. It is all dirt and it is very large."
```
Comment (lumbar527) [inline items.yaml:30]:
```suggestion
  description: "An old pair of pants. It is large and very dirty."
```

File: main.py [hunk 5]
```diff
@@ -243,6 +308,7 @@ def run(play):
     play = run(1)
 
 # finish up and save
+
```
Comment (lumbar527) [inline main.py:311]:
```suggestion
```

I don't really want this space between the comment and doing stuff.

File: start.yaml [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
 health: 10
+armor protection: .95
 held item: Sword
 inventory:
 - Sword
 - Healing Potion
-max health: 10
+- Leather Jacket
+- Leather Boots
+- Old Pants
```
Comment (lumbar527) [inline start.yaml:10]: Maybe you picked these up before the game starts, while you chase your friend into the dungeons.

**Review by lumbar527 — APPROVED** [review]
Mostly good!

**Comment by OcelotWalrus** [comment]
> Sure, though I'm not entirely sure by what you mean by this:

Basically adding a `swiftness` stat to armor pieces. More you have swiftness, more you have a chance in fights to 'dodge' enemies attacks (canceling enemies damages and print 'you doged the enemy attack

> Does that mean another random variable like enemy_max_health or enemy_damage that would be one of the enemies stats?

yes.

File: battle.py (hunk 1 above)
Comment (OcelotWalrus) [reply battle.py:78]:
```
round(random.uniform(0.50, 0.90), 1)
```

It generates a random floating number between `.5` and `.9` that will only have 1 number after the '.'.

File: items.yaml (hunk 3 above)
Comment (OcelotWalrus) [reply items.yaml:26]: Maybe. I'll make an other PR for the swiftness attribute I think.

File: main.py (hunk 5 above)
Comment (OcelotWalrus) [reply main.py:311]: Sure

File: start.yaml (hunk 6 above)
Comment (OcelotWalrus) [reply start.yaml:10]: Sorry, I actually added locally that but I don't know why it wasn't commited.

**Comment by OcelotWalrus** [comment]
Please I have stuff to do on this PR, so don't merge it for now.

**Comment by OcelotWalrus** [comment]
You can merge now.

File: main.py [hunk 7]
```diff
@@ -222,13 +342,94 @@ def run(play):
...
+        elif command.lower().startswith('p'):
+            # equipment
+            if player["held item"] == " ":
+                print("You are currently holding no weapon")
+            else:
+                print("You are holding a/an " + COLOR_RED + player["held item"] + COLOR_RESET_ALL)
+            if player["held chestplate"] == " ":
+                print("You are currently holding no chestplate")
+            else:
+                print("You are holding a/an " + COLOR_RED + player["held chestplate"] + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline main.py:395]:
```suggestion
                print("You are wearing a/an " + COLOR_RED + player["held chestplate"] + COLOR_RESET_ALL)
```
And for below.

File: items.yaml [hunk 8]
```diff
@@ -3,6 +3,10 @@ Sword:
   damage: 5
   defend: 6
   description: "Your generic sword."
+Bow:
+  type: "Weapon"
+  damage: 4
+  defend: 9
 Healing Potion:
```
Comment (lumbar527) [inline items.yaml:10]:
```suggestion
  description: "A bow. Perfectly normal. You use it to shoot arrows."
Healing Potion:
```

File: main.py [hunk 9]
```diff
@@ -3,6 +3,7 @@
 import pickle
 import battle
 import os
+import keyboard
```
Comment (lumbar527) [inline main.py:6]:
```suggestion
```

Is this used?

**Review by lumbar527 — APPROVED** [review]
I'm liking this, just a few things. I would like `agility`/`swiftness` attributes — do you want to add that here or in another PR?

**Comment by OcelotWalrus** [comment]
> I'm liking this, just a few things. I would like agility/swiftness attributes — do you want to add that here or in another PR?

I'll add it in the next PR where I'll add enemies stuff we discussed later and agility too.

**Comment by lumbar527** [comment]
> > I'm liking this, just a few things. I would like agility/swiftness attributes — do you want to add that here or in another PR?
> 
> I'll add it in the next PR where I'll add enemies stuff we discussed later and agility too.

OK

File: main.py (hunk 9 above)
Comment (OcelotWalrus) [reply main.py:6]: No, it was for a test.

**Comment by OcelotWalrus** [comment]
You can marge it.

File: main.py [hunk 10]
```diff
@@ -206,13 +299,39 @@ def run(play):
...
+            print(" ")
+            # equipment
+            if player["held item"] == " ":
+                print("You are currently holding no weapon")
+            else:
+                print("You are holding a/an " + COLOR_RED + player["held item"] + COLOR_RESET_ALL)
+            if player["held chestplate"] == " ":
+                print("You are currently holding no chestplate")
+            else:
+                print("You are holding a/an " + COLOR_RED + player["held chestplate"] + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline main.py:316]:
```suggestion
                print("You are wearing a/an " + COLOR_RED + player["held chestplate"] + COLOR_RESET_ALL)
```

**Review by lumbar527 — APPROVED** [review]
You still need to change all of these things (there's more than the review comments show). I'll merge soon.

**Comment by OcelotWalrus** [comment]
> You still need to change all of these things (there's more than the review comments show). I'll merge soon.

I've commited that.

File: start.yaml [hunk 11]
```diff
@@ -18,6 +18,7 @@ max health: 15
 x: 0
 xp: 0
 y: 0
+map_zone: "Kathallion Plains"
```
Comment (lumbar527) [inline start.yaml:21]: You're in a dungeon

Comment (OcelotWalrus) [reply start.yaml:21]: But before entering the dungeon, you're outside right?

Comment (lumbar527) [reply start.yaml:21]: Yes, but you enter the dungeon before you start playing

Comment (lumbar527) [reply start.yaml:21]: Also, I've been using spaces in my data file variables:
```suggestion
map zone: "Dungeon Entrance"
```

Comment (OcelotWalrus) [reply start.yaml:21]: I'll fix that later.

Comment (OcelotWalrus) [reply start.yaml:21]: I fixed that by commit aa666c7.

File: start.yaml [hunk 12]
```diff
@@ -1,12 +1,25 @@
...
-max health: 10
+- Leather Jacket
+- Leather Boots
+- Old Pants
+- Stretched Bag
+max health: 15
 x: 0
 xp: 0
 y: 0
+map_zone: "Kathallion Plains"
```
Comment (lumbar527) [inline start.yaml:21]:
```suggestion
map zone: "Dungeon Entrance"
```
You forgot to do this one.

**Comment by lumbar527** [comment]
Merge ready?

**Comment by OcelotWalrus** [comment]
Yes, thanks!

**Comment by lumbar527** [comment]
Merge canceled.

**Comment by OcelotWalrus** [comment]
> Merge canceled.

Sorry about that, you can merge now.

**Pull request merged**